extractor/src/main/java/org/schabi/newpipe/extractor/stream/VideoStream.java

```diff
@@ -61,86 +135,214 @@ public VideoStream(String url, boolean isVideoOnly, ItagItem itag) {
         this.width = itag.getWidth();
         this.quality = itag.getQuality();
         this.fps = itag.fps;
-    }
-
-    public VideoStream(String url, String torrentUrl, MediaFormat format, String resolution) {
-        this(url, torrentUrl, format, resolution, false);
-    }
-
-    public VideoStream(String url, String torrentUrl, MediaFormat format, String resolution, boolean isVideoOnly) {
-        super(url, torrentUrl, format);
         this.resolution = resolution;
         this.isVideoOnly = isVideoOnly;
     }
 
+    /**
+     * Create a new video stream.
+     * <p>
+     * The media format and the resolution will be set by using respectively
+     * {@link ItagItem#getMediaFormat()} and {@link ItagItem#resolutionString}.
+     * </p>
+     *
+     * @param id          the ID which uniquely identifies the stream, e.g. for YouTube this would
+     *                    be the itag
+     * @param url         the URL of the stream
+     * @param isVideoOnly whether the stream is video-only
+     * @param itag        the {@link ItagItem} corresponding to the stream, which cannot be null
+     */
+    public VideoStream(final String id,
+                       final String url,
+                       final boolean isVideoOnly,
+                       @Nonnull final ItagItem itag) {
+        this(id, url, itag.getMediaFormat(), itag.resolutionString, isVideoOnly);
+        this.itagItem = itag;
+        this.itag = itag.id;
+        this.bitrate = itag.getBitrate();
+        this.initStart = itag.getInitStart();
+        this.initEnd = itag.getInitEnd();
+        this.indexStart = itag.getIndexStart();
+        this.indexEnd = itag.getIndexEnd();
+        this.codec = itag.getCodec();
+        this.height = itag.getHeight();
+        this.width = itag.getWidth();
+        this.quality = itag.getQuality();
+        this.fps = itag.fps;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    public boolean equalStats(Stream cmp) {
-        return super.equalStats(cmp) && cmp instanceof VideoStream &&
-                resolution.equals(((VideoStream) cmp).resolution) &&
-                isVideoOnly == ((VideoStream) cmp).isVideoOnly;
+    public boolean equalStats(final Stream cmp) {
+        return super.equalStats(cmp) && cmp instanceof VideoStream
+                && resolution.equals(((VideoStream) cmp).resolution)
+                && isVideoOnly == ((VideoStream) cmp).isVideoOnly;
     }
```

Comment:
This method does the same as `equals`, so it could probably be removed, but it also makes sense to keep it since it takes a `Stream` instead of an `Object` and is used in specific cases. (**do not change anything**)